File: forkjoin-apikit-example/src/main/java/org/forkjoin/apikit/example/api/AccountApiController.java
package org.forkjoin.apikit.example.api;

import org.forkjoin.apikit.core.Account;
import org.forkjoin.apikit.core.Api;
import org.forkjoin.apikit.core.Result;
import org.forkjoin.apikit.example.ExampleAccount;
import org.forkjoin.apikit.example.model.User;
import org.forkjoin.apikit.example.services.AccountService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

/**
 * @author zuoge85 on 15/6/11.
 */
//@Api
@RestController
@RequestMapping(value = "v1")
public class AccountApiController {

    @Autowired
    private AccountService accountService;

    /**
     * @return 返回授权token
     */
    @RequestMapping(value = "account/login", method = RequestMethod.POST)
    @Account(false)
    public String login() throws Exception {
        String token = accountService.login();
        return token;
    }

    /**
     * 测试需要登录5
     */
    @RequestMapping(value = "account/testLogin", method = {RequestMethod.POST})
    public Result<Void> testLogin(ExampleAccount account) throws Exception {
        return Result.createSuccess();
    }

    /**
     * 测试不需要登录
     */
    @Account(false)
    @RequestMapping(value = "baseUrl/testNotLogin", method = {RequestMethod.GET, RequestMethod.POST})
    public Result<User> testNotLogin() throws Exception {
        return Result.createSuccess();
    }
}

File: forkjoin-apikit-example/src/test/js/api/AccountApiController.js

import {AbstractApi} from 'apikit-core'

import {requestGroupImpi} from 'apikit-core'


class AccountApiController extends AbstractApi {

    /**
        *  返回授权token
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>account/login</b>
    * <ul>
    * <li><b>Model:</b> string</li>
    * </ul>
    * </div>
    * @see string

     */
    login = () => {
        let _path = null;
        return super._request("test", "accountApiController", "POST", "account/login", _path, null);
    }


    /**
        * 测试需要登录5
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>account/testLogin</b>
    * <ul>
    * <li><b>Model:</b> void</li>
    * <li>需要登录</li>
    * </ul>
    * </div>

     */
    testLogin = () => {
        let _path = null;
        return super._request("test", "accountApiController", "POST", "account/testLogin", _path, null);
    }


    /**
        * 测试不需要登录
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>baseUrl/testNotLogin</b>
    * <ul>
    * <li><b>Model:</b> User</li>
    * </ul>
    * </div>
    * @see User

     */
    testNotLogin = () => {
        let _path = null;
        return super._request("test", "accountApiController", "GET", "baseUrl/testNotLogin", _path, null);
    }

}

export { AccountApiController };
const accountApiController = new AccountApiController();
accountApiController._init(requestGroupImpi);
export default accountApiController;

